akka-javasdk/src/main/scala/akka/javasdk/impl/eventsourcedentity/EventSourcedEntityImpl.scala

```diff
@@ -0,0 +1,292 @@
+import akka.javasdk.eventsourcedentity.EventSourcedEntityContext
+import akka.javasdk.impl.AbstractContext
+import akka.javasdk.impl.ActivatableContext
+import akka.javasdk.impl.AnySupport
+import akka.javasdk.impl.ComponentDescriptor
+import akka.javasdk.impl.EntityExceptions
+import akka.javasdk.impl.EntityExceptions.EntityException
+import akka.javasdk.impl.ErrorHandling.BadRequestException
+import akka.javasdk.impl.JsonMessageCodec
+import akka.javasdk.impl.MetadataImpl
+import akka.javasdk.impl.Settings
+import akka.javasdk.impl.effect.ErrorReplyImpl
+import akka.javasdk.impl.effect.MessageReplyImpl
+import akka.javasdk.impl.effect.NoSecondaryEffectImpl
+import akka.javasdk.impl.eventsourcedentity.EventSourcedEntityEffectImpl.EmitEvents
+import akka.javasdk.impl.eventsourcedentity.EventSourcedEntityEffectImpl.NoPrimaryEffect
+import akka.javasdk.impl.eventsourcedentity.EventSourcedEntityRouter.CommandHandlerNotFound
+import akka.javasdk.impl.eventsourcedentity.EventSourcedEntityRouter.EventHandlerNotFound
+import akka.javasdk.impl.telemetry.SpanTracingImpl
+import akka.javasdk.impl.telemetry.Telemetry
+import akka.runtime.sdk.spi.SpiEntity
+import akka.runtime.sdk.spi.SpiEventSourcedEntity
+import akka.runtime.sdk.spi.SpiSerialization
+import akka.runtime.sdk.spi.SpiSerialization.Deserialized
+import com.google.protobuf.ByteString
+import com.google.protobuf.any.{ Any => ScalaPbAny }
+import io.grpc.Status
+import io.opentelemetry.api.trace.Span
+import io.opentelemetry.api.trace.Tracer
+import org.slf4j.LoggerFactory
+import org.slf4j.MDC
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[impl] object EventSourcedEntityImpl {
+  private val log = LoggerFactory.getLogger(this.getClass)
+
+  private class CommandContextImpl(
+      override val entityId: String,
+      override val sequenceNumber: Long,
+      override val commandName: String,
+      override val commandId: Long, // FIXME remove
+      override val isDeleted: Boolean,
+      override val metadata: Metadata,
+      span: Option[Span],
+      tracerFactory: () => Tracer)
+      extends AbstractContext
+      with CommandContext
+      with ActivatableContext {
+    override def tracing(): Tracing = new SpanTracingImpl(span, tracerFactory)
+  }
+
+  private class EventSourcedEntityContextImpl(override final val entityId: String)
+      extends AbstractContext
+      with EventSourcedEntityContext
+
+  private final class EventContextImpl(entityId: String, override val sequenceNumber: Long)
+      extends EventSourcedEntityContextImpl(entityId)
+      with EventContext
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[impl] final class EventSourcedEntityImpl[S, E, ES <: EventSourcedEntity[S, E]](
+    configuration: Settings,
+    tracerFactory: () => Tracer,
+    componentId: String,
+    componentClass: Class[_],
+    messageCodec: JsonMessageCodec,
+    factory: EventSourcedEntityContext => ES,
+    snapshotEvery: Int)
```

Comment:
This isn't used here, passed to runtime via settings instead. I'll follow up in separate PR.